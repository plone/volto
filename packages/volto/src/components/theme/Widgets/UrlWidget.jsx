import React from 'react';
import cx from 'classnames';
import { UniversalLink } from '@plone/volto/components';

const UrlWidget = ({ value, children, className }) =>
  value ? (
    <UniversalLink href={value} className={cx(className, 'url', 'widget')}>
      {children ? children(value) : value}
    </UniversalLink>
  ) : (
    ''
  );

export default UrlWidget;
