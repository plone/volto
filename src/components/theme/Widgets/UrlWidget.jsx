import React from 'react';
import cx from 'classnames';

const UrlWidget = ({ value, children, className }) =>
  value ? (
    <a
      href={value}
      className={cx(className, 'url', 'widget')}
      rel="noreferrer"
      target="_blank"
    >
      {children ? children(value) : value}
    </a>
  ) : (
    ''
  );

export default UrlWidget;
