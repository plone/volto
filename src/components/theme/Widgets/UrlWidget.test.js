import React from 'react';
import renderer from 'react-test-renderer';
import UrlWidget from './UrlWidget';

describe('UrlWidget', () => {
  it('renders an empty URL view widget component', () => {
    const component = renderer.create(<UrlWidget />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders an URL view widget component', () => {
    const component = renderer.create(
      <UrlWidget className="metadata" value="http://foobar.com" />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders an URL view widget component with children', () => {
    const component = renderer.create(
      <UrlWidget className="metadata" value="http://foobar.com">
        {(child) => <strong>{child}</strong>}
      </UrlWidget>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
