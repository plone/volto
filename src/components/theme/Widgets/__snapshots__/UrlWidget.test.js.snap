// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UrlWidget renders an URL view widget component 1`] = `
<a
  className="external metadata url widget"
  href="http://foobar.com"
  rel="noopener noreferrer"
  target="_blank"
  title={null}
>
  http://foobar.com
</a>
`;

exports[`UrlWidget renders an URL view widget component with children 1`] = `
<a
  className="external metadata url widget"
  href="http://foobar.com"
  rel="noopener noreferrer"
  target="_blank"
  title={null}
>
  <strong>
    http://foobar.com
  </strong>
</a>
`;

exports[`UrlWidget renders an empty URL view widget component 1`] = `""`;
